snooty/postprocess.py

```diff
@@ -213,31 +212,20 @@ def _attach_doc_title(self, filename: FileId, node: n.RefRole) -> None:
             )
             return
 
-        # Add .txt to end of doc role target path
-        target_path = PurePath(node.fileid)
-        # Adding the current suffix first takes into account dotted targets
-        new_suffix = target_path.suffix + ".txt"
-        target_path = target_path.with_suffix(new_suffix)
-
         relative, _ = util.reroot_path(
-            target_path, filename, self.project_config.source_path
+            FileId(node.fileid), filename, self.project_config.source_path
```

Comment:
fileid is already a FileId, right? Nice thing about FileId is that it's immutable, so you never need to copy it for any purpose!